code/go/0chain.net/smartcontract/storagesc/models.go

```diff
@@ -862,6 +862,54 @@ func (sa *StorageAllocation) removeBlobber(
 			sa.BlobberDetails[i] = sa.BlobberDetails[len(sa.BlobberDetails)-1]
 			sa.BlobberDetails = sa.BlobberDetails[:len(sa.BlobberDetails)-1]
 			removedBlobber.Used -= d.Size
+
+			if d.ChallengePartitionLoc != nil {
```

Comment:
Move the code block in the for loop for removing a blobber into a separate function or method to keep single method short.

Also, we'd better wraps those partitions related actions in blobber's own methods, and call them to handle the business logic.